test/testsuite/lantest_io_monitor.c

```diff
@@ -0,0 +1,675 @@
+
+    return found.pids[0];
+}
+
+/* Read IO statistics from /proc_io */
+static int read_proc_io(pid_t pid, unsigned long long *read_ops,
+                        unsigned long long *write_ops)
+{
+    FILE *io_file;
+    char io_path[256];
+    char line[256];
+    int found_read = 0, found_write = 0;
+    *read_ops = 0;
+    *write_ops = 0;
+    snprintf(io_path, sizeof(io_path), "/proc_io/%d/io", pid);
+    io_file = fopen(io_path, "r");
+
+    if (!io_file) {
+        fprintf(stderr, "Cannot open %s (errno: %d)\n", io_path, errno);
+        return 0;
+    }
+
+    while (fgets(line, sizeof(line), io_file) && (!found_read || !found_write)) {
+        if (sscanf(line, "syscr: %llu", read_ops) == 1) {
+            found_read = 1;
+        } else if (sscanf(line, "syscw: %llu", write_ops) == 1) {
+            found_write = 1;
+        }
+    }
+
+    fclose(io_file);
+
+    if (Debug) {
+        fprintf(stderr, "DEBUG: IO result (%d) - syscr: %llu, syscw: %llu\n", pid,
+                *read_ops, *write_ops);
+    }
+
+    return (found_read && found_write) ? 0 : -1;
+}
+
+/* Capture IO values - consolidated function for start and stop */
+void capture_io_values(int is_start)
+{
+    if (!io_monitoring_enabled) {
+        return;
+    }
+
+    /* Process afpd_pid */
+    if (afpd_pid > 0) {
+        unsigned long long *reads = is_start ? &afpd_start_reads : &afpd_end_reads;
+        unsigned long long *writes = is_start ? &afpd_start_writes : &afpd_end_writes;
+
+        if (read_proc_io(afpd_pid, reads, writes) != 0) {
+            *reads = *writes = 0;
+        }
+    }
+
+    /* Process cnid_dbd_pid */
+    if (cnid_dbd_pid > 0) {
+        unsigned long long *reads = is_start ? &cnid_start_reads : &cnid_end_reads;
+        unsigned long long *writes = is_start ? &cnid_start_writes : &cnid_end_writes;
+
+        if (read_proc_io(cnid_dbd_pid, reads, writes) != 0) {
+            *reads = *writes = 0;
+        }
+    }
+}
+
+/* Get IO difference for specific PID - consolidated for read and write */
+unsigned long long iodiff_io(pid_t pid, int is_write)
+{
+    unsigned long long start_val, end_val;
+
+    if (!io_monitoring_enabled || pid == 0) {
+        return 0;  /* Return 0 instead of -1 for unsigned return type */
+    }
+
+    if (pid == afpd_pid) {
+        start_val = is_write ? afpd_start_writes : afpd_start_reads;
+        end_val = is_write ? afpd_end_writes : afpd_end_reads;
+    } else if (pid == cnid_dbd_pid) {
+        start_val = is_write ? cnid_start_writes : cnid_start_reads;
+        end_val = is_write ? cnid_end_writes : cnid_end_reads;
+    } else {
+        return 0;
+    }
+
+    return (end_val > start_val) ? end_val - start_val : 0;
+}
+
+/* Safe PID string parsing helper */
+static pid_t safe_parse_pid(const char *pid_str)
+{
+    char *endptr;
+    long val;
+    /* Define PID_MAX if not available from system headers */
+#ifndef PID_MAX
+    /* Conservative default maximum PID value */
+#define PID_MAX 99999
+#endif
```

Comment:
I'm generally favoring having global constants defined in the headers rather than inline in the functions where they're used.

Over the years I've found the same or similar constants defined multiple times across the codebase, and I've made an effort of consolidating them in appropriate headers.

But this is not a strict rule; if you think this one makes more sense as a one-off local constant here that is reasonable too.